lib/GenomeFileUtil/core/GenomeInterface.py

```diff
@@ -128,54 +171,70 @@ def get_one_genome(self, params):
-        else:
-            if params.get('upgrade') or 'feature_counts' not in data:
-                data = self._update_genome(data)
-
-        # check all handles point to shock nodes owned by calling user
-        self._own_handle(data, 'genbank_handle_ref')
-        self._own_handle(data, 'gff_handle_ref')
-        if "AnnotatedMetagenomeAssembly" not in ws_datatype:
-            self._check_dna_sequence_in_features(data)
-            data['warnings'] = self.validate_genome(data)
-
-        # sort data
-        data = GenomeUtils.sort_dict(data)
-        # dump genome to scratch for upload
-        data_path = os.path.join(self.scratch, name + ".json")
-        json.dump(data, open(data_path, 'w'))
-        if 'hidden' in params and str(params['hidden']).lower() in ('yes', 'true', 't', '1'):
-            hidden = 1
-        else:
-            hidden = 0
+    def _save_genome_mass(self, params, validate_genome=True):
 
-        if isinstance(workspace, int) or workspace.isdigit():
-            workspace_id = workspace
-        else:
-            workspace_id = self.dfu.ws_name_to_id(workspace)
-
-        save_params = {'id': workspace_id,
-                       'objects': [{'type': ws_datatype,
-                                    'data_json_file': data_path,
-                                    'name': name,
-                                    'meta': meta,
-                                    'hidden': hidden}]}
-        dfu_oi = self.ws_large_data.save_objects(save_params)[0]
-        returnVal = {'info': dfu_oi, 'warnings': data.get('warnings', [])}
-        return returnVal
+        workspace_id = params[_WSID]
+        inputs = params[_INPUTS]
+
+        objects = []
+        warnings = []
+
+        for input_params in inputs:
+
+            obj = {}
+
+            # retrive required params
+            name = input_params['name']
+            data = dict(input_params['data'])
+
+            # XXX there is no `workspace_datatype` param in the spec
+            # NOTE: This allows a user to specify any arbitrary workspace type which could cause,
+            # in the worst case, data corruption. It should be removed from the API
+            # (note it is not currently documented) so users cannot access it.
+            ws_datatype = input_params.get('workspace_datatype', "KBaseGenomes.Genome")
+            # XXX there is no `meta` param in the spec
+            meta = input_params.get('meta', {})
+
+            obj["type"] = ws_datatype
+            obj["name"] = name
+            obj["meta"] = meta
+
+            if "AnnotatedMetagenomeAssembly" not in ws_datatype:
+                if input_params.get('upgrade') or 'feature_counts' not in data:
+                    data = self._update_genome(data)
+
+            # check all handles point to shock nodes owned by calling user
+            self._own_handle(data, 'genbank_handle_ref')
+            self._own_handle(data, 'gff_handle_ref')
```

Comment:
This could also fail, and given for our use case the handles are guaranteed to be owned by the user (right?) seems like we could add a toggle to skip this step IIUC. Document for now

Reply:
The second part of this test appears to be specifically designed to test behavior with two different users. In fact, two distinct test tokens are passed in CI just for this purpose. Are these related to the discussion we're having here?

https://github.com/kbaseapps/GenomeFileUtil/blob/4819598e65eea38d3c38d8c2ab808aca0d9697fd/test/problematic_tests/save_genome_test.py#L222

Reply:
It seems to me the answer to that would be based on whether the new method this project has added to the GFU public API allows for specifying a handle that already exists, unless I'm missing something

Reply:
The only new public API introduced in this project is the GTGs method, which is essentially a batch version of GTG. So to me, the question becomes whether the original codebase allows specifying an existing handle.

Reply:
So the code we added / modified doesn't deal with pre-existing handles?

Reply:
The GTG input does not include handle information: https://github.com/kbaseapps/GenomeFileUtil/blob/c306fc6f121abd1d548d69a467d18173c2b34163/GenomeFileUtil.spec#L71

we generate our own handle using the following function:
https://github.com/kbaseapps/GenomeFileUtil/blob/c306fc6f121abd1d548d69a467d18173c2b34163/lib/GenomeFileUtil/core/GenbankToGenome.py#L286

Reply:
:+1: